5_Smallest_Multiple.py

```diff
@@ -0,0 +1,24 @@
+# this takes forever to run, it isnt optimized
```

Comment:
Yep, the naive solution is _slow_! There is a math trick here. For example, if you want N to be divisible by 10, you've already guaranteed it's divisible by 2 and 5. So using the example from the problem of 1-10:

```
10 = 2 * 5
9 = 3 * 3
8 = 2 * 2 * 2
7 = 7
6  = 2 * 3
5 = 5
4 = 2 * 2
3 = 3
2 = 1
1 = 1
```

So you see there is a lot of overlap. What you really need are the following, and counts:
* 2: need 3, b/c 8
* 3: need 2, b/c 9
* 5: need 1
* 7: need 1

And magically, 2^3 * 3^2 * 5 * 7 = 2520